### PR TITLE
Add BlobMeta.deleted_on

### PR DESCRIPTION
**Pull request by millerdev**
This lays the groundwork for retention of deleted non-temporary blob metadata. This needs to be deployed and indexes created on large environments before the rest of the implementation can be rolled out.

@mkangia @dannyroberts

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]

File: corehq/form_processor/management/commands/run_sql.py [hunk 1]
```diff
@@ -497,7 +497,15 @@ def meta_fields(meta, with_variable_fields=True):
 """)
 
 
+# see https://github.com/dimagi/commcare-hq/pull/22932
+blobmeta_partial_key = """
+CREATE UNIQUE INDEX CONCURRENTLY IF NOT EXISTS blobs_blobm_key_14b5b7_partial
+ON public.blobs_blobmeta (key) WHERE deleted_on IS NULL;
```
Comment (mkangia) [inline corehq/form_processor/management/commands/run_sql.py]: you meant to keep it "IS NULL"?

Comment (millerdev) [reply corehq/form_processor/management/commands/run_sql.py]: Yes, it excludes deleted rows from the index.

**Comment by millerdev** [comment]
The partial index and more complicated queries is making me reconsider this implementation. I'm exploring creating a separate table into which deleted non-temporary blob metadata will be copied. This will keep other (non-delete) blob metadata queries simple and should require less storage space since the deleted records do not need the other indexes maintained on this table.

**Comment by millerdev** [comment]
Thought of a better way to do this: https://github.com/dimagi/commcare-hq/pull/22969
No new indexes needed.

File: corehq/form_processor/management/commands/run_sql.py (hunk 1 above)
Comment (mkangia) [reply corehq/form_processor/management/commands/run_sql.py]: a deleted row would have `deleted_on` set and hence this should be `NOT NULL`?
fine to ignore it if its not relevant anymore

**Pull request closed without merging**